### PR TITLE
Migration of account_cutoff modules to 10.0

### PR DESCRIPTION
**Pull request by adrienpeiffer**
- [x] Forward modification on https://github.com/OCA/account-closing/pull/37

### CODE REVIEW

File: setup/account_cutoff_base/.eggs/README.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+This directory contains eggs that were downloaded by setuptools to build, test, and run plug-ins.
```
Comment (pedrobaeza) [inline setup/account_cutoff_base/.eggs/README.txt:1]: Why this file?

File: setup/account_cutoff_base/odoo/addons/account_cutoff_base [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../../account_cutoff_base
```
Comment (pedrobaeza) [inline setup/account_cutoff_base/odoo/addons/account_cutoff_base:1]: And these?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: account_cutoff_base/models/account_cutoff.py [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+class AccountCutoff(models.Model):
+    _name = 'account.cutoff'
+    _rec_name = 'cutoff_date'
+    _order = 'cutoff_date desc'
+    _inherit = ['mail.thread']
+    _description = 'Account Cut-off'
+
+    @api.multi
+    @api.depends('line_ids')
```
Comment (Garamotte) [inline account_cutoff_base/models/account_cutoff.py]: I think this field should also depend on `line_ids.cutoff_amount`.

File: account_cutoff_base/security/account_cutoff_base_security.xml [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
-    <data noupdate="0">
-        <record id="account_cutoff_multi_company_rule" model="ir.rule">
-            <field name="name">Account Cutoff Multi Company</field>
-            <field name="model_id" ref="model_account_cutoff"/>
-            <field name="global" eval="True"/>
-            <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
-        </record>
-    </data>
-</openerp>
+<odoo noupdate="1">
```
Comment (Garamotte) [inline account_cutoff_base/security/account_cutoff_base_security.xml:2]: Is the change from `noupdate="0"` to `noupdate="1"` intentional ?

File: account_cutoff_prepaid/models/account_cutoff.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+from odoo.exceptions import UserError, ValidationError
+
+
+class AccountCutoff(models.Model):
+    _inherit = 'account.cutoff'
+
+    @api.model
+    def _get_default_source_journals(self):
+        res = []
+        type = self._context.get('type')
```
Comment (Garamotte) [inline account_cutoff_prepaid/models/account_cutoff.py]: Please avoid using python's reserved names as variable name.

File: account_cutoff_prepaid/models/account_cutoff.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        # Loop on selected account move lines to create the cutoff lines
+        for aml in amls:
+            line_obj.create(self._prepare_prepaid_lines(aml, mapping))
+        return True
+
+    @api.model
+    def _inherit_default_cutoff_account_id(self):
+        account_id = super(AccountCutoff, self).\
+            _inherit_default_cutoff_account_id()
+        type = self._context.get('type')
```
Comment (Garamotte) [inline account_cutoff_prepaid/models/account_cutoff.py]: Please avoid using python's reserved names as variable name.

File: account_cutoff_prepaid/tests/test_account_cutoff_prepaid.py [hunk 7]
```diff
@@ -1,82 +1,78 @@
...
                 'start_date': self._date(start_date),
                 'end_date': self._date(end_date),
             })],
         })
-        workflow.trg_validate(self.uid, 'account.invoice', inv_id,
-                              'invoice_open', self.cr)
-        inv = self.inv_model.browse(self.cr, self.uid, inv_id)
-        self.assertEqual(amount, inv.amount_untaxed)
-        return inv_id
+        invoice.signal_workflow('invoice_open')
```
Comment (Garamotte) [inline account_cutoff_prepaid/tests/test_account_cutoff_prepaid.py]: The workflow has been removed, you must use `invoice.action_invoice_open()` now.

File: account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+                    'product_id':
+                    self.env.ref('product.product_product_17').id,
+                    'name': 'HD IPBX',
+                    'price_unit': 215.5,
+                    'quantity': 1,
+                    'account_id': self.account_revenue.id,
+                    }),
+                ],
+        })
+        invoice.signal_workflow('invoice_open')
```
Comment (Garamotte) [inline account_invoice_start_end_dates/tests/test_invoice_start_end_dates.py]: The workflow has been removed, you must use `invoice.action_invoice_open()` now.

**Review by Garamotte — CHANGES_REQUESTED** [review]
I think you should split some big python files into one file per model.

File: account_cutoff_base/security/account_cutoff_base_security.xml (hunk 4 above)
Comment (adrienpeiffer) [reply account_cutoff_base/security/account_cutoff_base_security.xml:2]: This change is also in the version 9.0.
For me, record rules can be set as no updatable like it's done in Odoo core.
@alexis-via Any comment about this change ?

Comment (pedrobaeza) [reply account_cutoff_base/security/account_cutoff_base_security.xml:2]: I agree. This way, you can set in your DB other rule if needed.

File: setup/account_cutoff_base/odoo/addons/account_cutoff_base (hunk 2 above)
Comment (adrienpeiffer) [reply setup/account_cutoff_base/odoo/addons/account_cutoff_base:1]: This is the symlink to the module in the namespace package

Comment (pedrobaeza) [reply setup/account_cutoff_base/odoo/addons/account_cutoff_base:1]: But why is this now needed?

File: account_cutoff_base/security/account_cutoff_base_security.xml (hunk 4 above)
Comment (Garamotte) [reply account_cutoff_base/security/account_cutoff_base_security.xml:2]: Ok, no problem.
I asked because this could have been a typo.

**Review by pedrobaeza — APPROVED** [review]

File: account_cutoff_base/security/account_cutoff_base_security.xml (hunk 4 above)
Comment (alexis-via) [reply account_cutoff_base/security/account_cutoff_base_security.xml:2]: yes, record rules should have noupdate="1"

**Review by atchuthan — APPROVED** [review]
Code review :+1: 
Tested functionality using PR https://github.com/OCA/sale-workflow/pull/357

**Review by dvdhinesh — APPROVED** [review]
Code :+1: 
Functional: account_invoice_start_end_dates :+1:

**Comment by pedrobaeza** [comment]
Can you squash last 2 commits before merging?

**Comment by adrienpeiffer** [comment]
@pedrobaeza it's done.

**Pull request merged**